configs/provider_meta.go

```diff
@@ -13,8 +13,20 @@ type ProviderMeta struct {
 }
 
 func decodeProviderMetaBlock(block *hcl.Block) (*ProviderMeta, hcl.Diagnostics) {
+	// provider_meta must be a static map. We can verify this by attempting to
+	// evaluate the values.
+	attrs, diags := block.Body.JustAttributes()
+	if diags.HasErrors() {
+		return nil, diags
```

Comment:
What diags are returned here? Does it help the user know how to fix their problem?

Reply:
It's an `hcl.Diagnostics`, so will contain the source reference. This shouldn't be seen by endusers, as its use needs to be coordinated with the provider itself (and is currently only used by Google). Validating will allow the module developers to catch the problem before publishing the module.